jetty-ant/src/main/java/org/eclipse/jetty/ant/AntWebInfConfiguration.java

```diff
@@ -88,7 +89,7 @@ public void matched(URI uri) throws Exception
                             }
                             catch (URISyntaxException e)
                             {
-                                containerUris[i] = new URI(u.toString().replaceAll(" ", "%20"));
+                                containerUris[i] = new URI(URIUtil.encodeSpaces(u.toString()));
```

Comment:
why are we just encoding spaces? and then only on errors?   This code looks optimistic at best!

Reply:
Both Apache Ant and Eclipse [Equinox OSGi](https://github.com/eclipse/jetty.project/issues/2575) do horrible things with URL.
We often get handed unencoded URLs and are expected to use them.
The try, then light touch encode on exception are a "best effort" for us in these situations.

Reply:
Note that the second `new URI(URIUtil.encodeSpaces(` line will throw a URISyntaxException out of that method if the input is horribly bad.